app/View/Services/frmServices.cs

```diff
@@ -32,18 +32,29 @@ private void frmServices_Load(object sender, EventArgs e)
 
         private void btnSearch_Click(object sender, EventArgs e)
         {
-            //TODO: Populate the datagridview based on the filtered name provided in Search box.
-            if (!string.IsNullOrEmpty(txtSearch.Text) || !string.IsNullOrWhiteSpace(txtSearch.Text))
+            if (!string.IsNullOrEmpty(txtSearch.Text.Trim()))
             {
+
                 ServiceRepository repository = new ServiceRepository();
-                dgvServices.DataSource = repository.SearchService(txtSearch.Text);
+                dgvServices.DataSource = repository.SearchService(txtSearch.Text.Trim());
                 this.dgvServices.Columns["Id"].Visible = false;
             }
             else
             {
-                MessageBox.Show("The search field is empty, please provide.", "Empty field", MessageBoxButtons.OK, MessageBoxIcon.Information);
+                RefreshDataGridView();
```

Comment:
kindly retain the message dialog to inform user that the search text field is empty.